pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -49,6 +50,19 @@ func (r *Client) CreateSnapshot(vmRef ref.Ref) (snapshot string, err error) {
 		return
 	}
 	snapsService := vmService.SnapshotsService()
+
+	locked, err := r.isAnySnapshotLocked(snapsService)
```

Comment:
yeah, that's a possible approach - but wouldn't it make more sense to initiate the operation and try to address the case that the operation failed because of a lock more broadly so we'll also over the case that the vm is, e.g., being migrated?

Reply:
I'll check it out but afaik even when the VM is being migrated it does not create the snapshot at the same plus it looks like the oVirt engine allows just one operation on the VM snapshots at once.

Reply:
that's right
the thing is that originally oVirt used locks at the database level, like that status of snapshots, and in-memory locks where used for a very limited time - they were acquired when a command started, validated and then they were released when the database-level locks were set. for quite a few years oVirt uses in-memory lock differently - for the whole duration of the command execution. we  still kept the database-level locks (for various reasons - presentation in the UI, tools that query the database directly, etc) but in theory in most-flows they are pretty much redundant